docs/fundamentals-and-concepts/stellar-data-structures/contracts.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: Smart Contracts
+sidebar_position: 50
+---
+
+A smart contract is a programmed set of executable code and state that can be invoked or used on the Stellar network.
+
+## Wasm Bytecode
+
+Once a smart contract has been written by a developer and compiled into a WebAssembly (Wasm) executable file, it can then be "installed" onto the Stellar network. A `CONTRACT_DATA` [ledger entry](ledgers.mdx) is created to store this binary data and its unique identifier is the hash of the executable file. This binary executable is stored independently from its deployed contract(s). When a Stellar transaction attempts to invoke a contract function, the first thing to take place is this Wasm bytecode is retrieved from the ledger and a secure, isolated runtime virtual machine ("VM") is instantiated so it can run the bytecode for the contract and then exit.
+
+## Contract Instances
+
+After the executable bytecode is installed on-chain, contract instances can be deployed which reference the afformentioned bytecode. A smart contract executable can have a one-to-many relationship with "contract instances" which function independantly. This means the same executable code can be used by multiple contract instances that all behave identically (because of the shared executable code), while maintaining seperate and distinct state data (because the data is tied to the contract instance). A contract instance is stored as its own ledger entry (**is that right? todo: fact-check this!**)
+
+```mermaid
+flowchart LR
+    A[my instance] & B[your instance]--> C[contract Wasm]
+```
+
+## Contract Storage
+
+In addition to the ledger entries that are created during the contract install/deploy process, each contract can create and access its own set of ledger entries. These ledger entries (as well as the contract code and the contract instance ledger entries) are subject to [state archival](https://soroban.stellar.org/docs/fundamentals-and-concepts/state-expiration) lifetimes (a ledger entry's "TTL ledger"). While they all function similarly, each type has its own fee and TTL behavior.
+
+### Temporary Storage
+
+- Cheapest fees.
+- Permanently deleted when its TTL ledger is reached, cannot be restored.
+- Suitable for time-bounded data (i.e. price oracles, signatures, etc.) and easily recreateable data.
+- Unlimited amount of storage.
+
+### Instance Storage
```

Comment:
It might be worth putting `instance storage` after `persistent storage` because it's more of a "meta" concept: under the hood, there's only temporary and persistent. As you state here `instance` is basically an abstracted convenience of basically saying "this should live as long as the contract instance" (which is persistent).

right, @SirTyson @dmkozh?

Reply:
As far as ledger keys and such, does `instance` storage have its own, separate from the contract instance? Like, could I still somehow ask RPC for the ledger key that corresponds to the contract's `Admin` address (that's been stored as instance storage)? Or, are _all_ the instance storage ledger entries just stored "alongside" or "with" (not sure the right word to use there) the contract instance itself?

Reply:
I swapped them around and added an `info` box. Is the info box too heavy-handed here?

Reply:
The instance 'storage' is just an accessor to the contract instance ledger entry (which is subject to the network limits, thus it can only take up to 64KB with current settings). So when you query the contract instance ledger key, the resulting ledger entry will contain every key-value pair stored in the instance storage. That said, it doesn't mean that API/client couldn't treat it in the same fashion as regular storage - it's mostly an interface choice. E.g. SDK interface presents instance storage as a separate storage type that has almost exactly the same interface as temp/persistent storage.

Reply:
Oooh, that's really cool to know. Thanks!